### PR TITLE
fix: obkv wal open

### PR DESCRIPTION
**Pull request by Rachelint**
## Rationale
In some cases(such as `Manifest::do_snapshot`), `actual_next_sequence` maybe less than `start_sequence`, and it will check this then return error in OBKV WAl.

In this pr, I refactor this open logic.

## Detailed Changes
Refactor wal open logic to allow `actual_next_sequence` less than `start_sequence`.

## Test Plan
Test by exist tests.

### CODE REVIEW

File: wal/src/kv_encoder.rs [hunk 1]
```diff
@@ -338,7 +338,9 @@ impl Decoder<MetaKey> for MetaKeyEncoder {
             }
         );
 
-        Ok(MetaKey { region_id })
+        Ok(MetaKey {
+            table_id: region_id,
```
Comment (jiacai2050) [inline wal/src/kv_encoder.rs:342]: Rename region_id to table_id

File: wal/src/rocks_impl/manager.rs [hunk 2]
```diff
@@ -463,8 +463,15 @@ impl RocksImpl {
             let meta_key = self.max_seq_meta_encoding.decode_key(iter.key())?;
             let meta_value = self.max_seq_meta_encoding.decode_value(iter.value())?;
             table_max_seqs
-                .entry(meta_key.region_id)
+                .entry(meta_key.table_id)
                 .and_modify(|v| {
+                    if meta_value.max_seq > *v {
+                        warn!(
+                            "RocksDB WAL found flushed_seq greater than actual_last_sequence,
```
Comment (jiacai2050) [inline wal/src/rocks_impl/manager.rs:470]: Add comments when this case will happen.

**Review by ShiKaiWi — APPROVED** [review]
LGTM

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**